be/src/fs/jindo/fs_jindo.cpp

```diff
@@ -0,0 +1,800 @@
+            jdo_freeHandleCtx(ctx);
+            jdo_destroyStore(store);
+            jdo_freeStore(store);
+            jdo_freeOptions(option);
+            return std::nullopt;
+        }
+        jdo_freeHandleCtx(ctx);
+
+        return std::make_shared<internal::BucketHost>(option, store);
+    }
+
+    std::map<std::string, std::string> default_option() const {
+        std::map<std::string, std::string> option;
+
+        auto load_from_core_site = [&option](const std::string& path) {
+            try {
+                boost::property_tree::ptree xml;
+                boost::property_tree::read_xml(path, xml, boost::property_tree::xml_parser::trim_whitespace);
+
+                if (auto maybe_properties = xml.get_child_optional("configuration")) {
+                    for (auto [name, property] : *maybe_properties) {
+                        // fail to use property.get("name") to work,
+                        // a walk around using pari
+                        std::pair<std::string, std::string> kv;
+                        for (auto [key, value] : property) {
+                            if (key == "name") {
+                                kv.first = value.data();
+                            } else if (key == "value") {
+                                kv.second = value.data();
+                            }
+                        }
+
+                        // copy all options ,but:
+                        // 1. fs.oss.accessKeyId
+                        // 2. fs.oss.accessKeySecret
+                        if (kv.first.compare("fs.oss.accessKeyId") == 0 ||
+                            kv.first.compare("fs.oss.accessKeySecret") == 0) {
+                            continue;
+                        }
+
+                        option.emplace(kv);
+                    }
+                }
+            } catch (boost::property_tree::xml_parser_error& error) {
+                JINDO_LOG_WARN << "fail to parse xml file:" << path << " reason:" << error.what();
+            }
+        };
+
+        // 1. from core-site.xml
+        auto home_conf = fmt::format("{}/conf/core-site.xml", boost::compute::detail::getenv("STARROCKS_HOME"));
+        if (fs::path_exist(home_conf)) {
+            JINDO_LOG_INFO << "using static config:" << home_conf;
+            load_from_core_site(home_conf);
+        } else {
+            // try from hadoop_confi_dir
+            for (auto candiate : strings::Split(boost::compute::detail::getenv("HADOOP_CONF_DIR"), ":")) {
+                auto conf = fmt::format("{}/core-site.xml", candiate.as_string());
+                if (fs::path_exist(conf)) {
+                    JINDO_LOG_INFO << "using static config:" << conf;
+                    load_from_core_site(conf);
+                    break;
+                }
+            }
+        }
+
+        // 2. cloud configuration
+        auto apply_cloud_configuration = [&option](const starrocks::TCloudConfiguration* cloud_configuration) {
```

Comment:
I think CloudConfiguration and hdfs_properties() actually do not worked.

Did you have test about CloudConfiguration and hdfs_properties() to authenticate?

Reply:
I only test the url and core-site one.
I assuming this two can embed customized key values like `fs.oss.xx` that latter extract when initializing `jdo_store`/`jdo_option`.
If it is not the case, surely removing it is fine.

Reply:
Actually, now CloudConfiguration and hdfs_properties() will not pass `fs.oss.xx` from FE.

But I hope you'll be able to support CloudConfiguration in this pr. It's easy. As for hdfs_properties(), you don't need to support it, it's not maintained anymore.

You can take a look at `AliyunCloudCredential.java` in FE, maybe just need to change `toThrift()` is enough.

In general, we do not encourage the use of `core-site.xml` to configuration authentication. We recommend specifying ak and sk (that is, CloudConfiguration) when creating a catalog.

Reply:
> Actually, now CloudConfiguration and hdfs_properties() will not pass `fs.oss.xx` from FE.
> 
> But I hope you'll be able to support CloudConfiguration in this pr. It's easy. As for hdfs_properties(), you don't need to support it, it's not maintained anymore.
> 
> You can take a look at `AliyunCloudCredential.java` in FE, maybe just need to change `toThrift()` is enough.
> 
I will take a look at it.

By the way, I was thinking hdfs_properties/CloudConfiguration will be a good place to inject dynamic context, but currently it is not that feasible to do so.(A proper plugin/extension point).

E.g. Internally,We have some concepts like virtual table/virtual warehouse or whatever it calls, and We dont want to expose sensitive data like password to user when they configurating table/datasource etc.So, in this case we are expecting to have some way to dynamic inject/resolve the tokens when generating execution plan by using some alias like meta data stored in hdfs_properties/CloudConfiguration.

> In general, we do not encourage the use of `core-site.xml` to configuration authentication. We recommend specifying ak and sk (that is, CloudConfiguration) when creating a catalog.

Reply:
Sorry, I missed one thing is that FE didn't introduce `jindo.jar` because of CVE problem, so even if you change CloudConfiguration's code, you still can't access jindo in FE.

And looks like `jindo.jar` isn't public in mvn repository, it's hard to introduce it.

What about just delete CloudConfiguration and hdfs_properties code here?

Reply:
I think another way is to translate/fallback/prefer `aliyun. oss. access_key` -> `fs.oss.accessKeyId`  like what `from_provided` in around line 629 dose?

Reply:
> Sorry, I missed one thing is that FE didn't introduce `jindo.jar` because of CVE problem, so even if you change CloudConfiguration's code, you still can't access jindo in FE.
> 
> And looks like `jindo.jar` isn't public in mvn repository, it's hard to introduce it.
> 
> What about just delete CloudConfiguration and hdfs_properties code here?

I believe we should have a well agreement on how `fs.oss.accessKeyId` is finally resolved.

Current approach is:
1. if ak/sk embeeded in path url, it is used as it is a clear user intention, or 
2. merge core-site/CloudConfiguration/hdfs_properties in to a unified kv store(special case to  ignore `fs.oss.accessKeyId` in `core-site` to avoid ambiguous/conflict).
3. query this kv store with key priority `fs.oss.accessKeyId` -> `fs.oss.bucket.${bucket}.accessKeyId`, which can be generalize/extend to be `fs.oss.accessKeyId` -> `fs.oss.bucket.${bucket}.accessKeyId` -> `aliyun.oss.access_key` -> `aws.s3.access_key`.

I had a quick read about the planner code path, and the be fs creation.
The driven context is `FSOption` which contributed to the 1 and mainly generated by catalog metadata, which should technically the only source of truth to working with `Jindo`.

But if leave it as the only source, we have no place to put tuning parameters into runtime without modifying the metadata(which I believe is a sufficient good constrain but works poorly in reality as in our case it required a DDL like privilege that implies escalation).
And if considering bring `core-site` for runtime behavior  tuner, should or should not filter `fs.oss.accessKeyId` out to keep the CloudConfiguration the only source of truth?

Reply:
resolved